backend/azure/dtmb/dtmb.go

```diff
@@ -0,0 +1,531 @@
+	completionToken := item.Properties["CompletionToken"].(string)
+	orchestrationName := item.Properties["OrchestrationName"].(string)
+	version := item.Properties["Version"].(string)
+
+	d.connectWorkerClientStream <- &dtmbprotos.ConnectWorkerClientMessage{
+		Message: &dtmbprotos.ConnectWorkerClientMessage_AbandonOrchestration{
+			AbandonOrchestration: &dtmbprotos.AbandonOrchestrationMessage{
+				OrchestrationId: string(item.InstanceID),
+				Name:            orchestrationName,
+				Version:         version,
+				CompletionToken: completionToken,
+			},
+		},
+	}
+	// this is a terminal event, so we can evict the cache
+	d.orchestrationHistoryCache.EvictCacheForOrchestrationID(string(item.InstanceID))
+	return nil
+}
+
+// GetActivityWorkItem gets a pending activity work item from the task hub or returns [ErrNoWorkItems]
+// if there are no pending activity work items.
+func (d *dtmb) GetActivityWorkItem(context.Context) (*backend.ActivityWorkItem, error) {
+	var ret *backend.ActivityWorkItem = nil
+	item := d.activityQueue.Dequeue()
+	if item == nil {
+		return nil, backend.ErrNoWorkItems
+	}
+
+	// this can't be the right way to provide the input, but how should it be done?
+
+	genericInputWrapper := protos.HistoryEvent{
+		EventType: &protos.HistoryEvent_GenericEvent{
+			GenericEvent: &protos.GenericEvent{
+				Data: &wrapperspb.StringValue{Value: string(item.GetInput())},
+			},
+		},
+	}
+
+	ret = &backend.ActivityWorkItem{
+		// SequenceNumber:  // how do I get the sequence number?
+		InstanceID: api.InstanceID(item.OrchestrationId),
+		NewEvent:   &genericInputWrapper,
+		Properties: map[string]interface{}{
+			"CompletionToken": item.GetCompletionToken(),
+			"ActivityName":    item.GetName(),
+		},
+	}
+	return ret, nil
+}
+
+// CompleteActivityWorkItem sends a message to the parent orchestration indicating activity completion.
+//
+// Returns [ErrWorkItemLockLost] if the work-item couldn't be completed due to a lock-lost conflict (e.g., split-brain).
+func (d *dtmb) CompleteActivityWorkItem(_ context.Context, item *backend.ActivityWorkItem) error {
+	bytes, err := backend.MarshalHistoryEvent(item.Result)
+	if err != nil {
+		return err
+	}
+	var failureDetails *dtmbprotos.FailureDetails = nil
+	if item.NewEvent.GetTaskFailed() != nil {
+		failureDetails = utils.ConvertTaskFailureDetails(item.NewEvent.GetTaskFailed().GetFailureDetails())
+	}
+
+	completionToken := item.Properties["CompletionToken"].(string)
+	activityName := item.Properties["ActivityName"].(string)
+
+	d.connectWorkerClientStream <- &dtmbprotos.ConnectWorkerClientMessage{
+		Message: &dtmbprotos.ConnectWorkerClientMessage_CompleteActivity{
+			CompleteActivity: &dtmbprotos.CompleteActivityMessage{
+				OrchestrationId: string(item.InstanceID),
+				Name:            activityName,    // what do I put here?
+				CompletionToken: completionToken, // what do I put here?
+				Result:          bytes,           // is this correct?
```

Comment:
It looks like you're sending a serialized history event? I assume that this is not what Alessandro's implementation is expecting. I think it should be the serialized data response from the activity function, which corresponds to `TaskCompletedEvent.GetResult().GetValue()`.

Reply:
Yeah that's also what I had before - but I kept changing things back and forth because of the type mismatch -- byte[] vs string. And the SQLite backend uses a serialized history event.

I'll change it back.